forte/predictor.py

```diff
@@ -0,0 +1,185 @@
+            if need_pad:
+                tensor, mask = converter.convert(features)
+                tensor_collection[tag]["tensor"] = tensor
+                tensor_collection[tag]["mask"] = mask
+            tensor_collection[tag]["features"] = features
+        return tensor_collection
+
+    def yield_batch(self, pack: DataPack) -> Dict:
+        for instance in pack.get(self.feature_resource['scope']):
+            feature_collection = {}
+            for tag, scheme in self.feature_resource['schemes'].items():
+                if scheme['type'] == DATA_INPUT:
+                    extractor = scheme["extractor"]
+                    feature = extractor.extract(pack, instance)
+                    feature_collection[tag] = feature
+
+            self.pack_pools.append(pack)
+            self.instance_pools.append(instance)
+            self.features_pools.append(feature_collection)
+
+            if len(self.features_pools) == self.batch_size:
+                yield self.convert(self.features_pools[:self.batch_size]), \
+                    self.pack_pools[:self.batch_size], \
+                    self.instance_pools[:self.batch_size]
+
+                self.features_pools.clear()
+                self.pack_pools.clear()
+                self.instance_pools.clear()
+
+        if not self.cross_pack and len(self.features_pools) > 0:
+            yield self.convert(self.features_pools), \
+                self.pack_pools, \
+                self.instance_pools
+            self.features_pools.clear()
+            self.pack_pools.clear()
+            self.instance_pools.clear()
+
+    def flush_batch(self) -> Dict:
+        if len(self.features_pools) > 0:
+            yield self.convert(self.features_pools), self.pack_pools, \
+                self.instance_pools
+            self.features_pools.clear()
+            self.pack_pools.clear()
+            self.instance_pools.clear()
+
+
+class Predictor(BaseProcessor):
+    '''This class will predict using the passed in model and add output
+    back to the datapack.
+    '''
+    def __init__(self, batch_size: int, model: Module,
+            predict_forward_fn: Callable, feature_resource: Dict,
+            cross_pack=True):
+        super().__init__()
+        self.feature_resource = feature_resource
+        self.batcher = Batcher(batch_size=batch_size,
+                                feature_resource=feature_resource,
+                                cross_pack=cross_pack)
+        self.model = model
+        self.predict_forward_fn = predict_forward_fn
+
+    def add_to_pack(self, predictions: Dict, packs: List[DataPack],
+                    instances: List[Annotation]):
+        for tag, preds in predictions.items():
+            for pred, pack, instance in zip(preds, packs, instances):
+                self.feature_resource['schemes'][tag]['extractor'].add_to_pack(
+                            pack, instance, pred)
+                pack.add_all_remaining_entries()
```

Comment:
Are we sure we want to call this function? Predictor is a BaseProcessor, and the system should do that.